src/account.ts

```diff
@@ -548,11 +535,62 @@ export class OpenZeppelinAccount extends Account {
         return signMultiCall(this.publicKey, this.keyPair, messageHash);
     }
 
+    public override async estimateDeployAccountFee(): Promise<starknet.FeeEstimation> {
+        this.assertNotDeployed();
+        const hre = await import("hardhat");
+
+        const contractFactory = await OpenZeppelinAccount.getContractFactory();
+        const classHash = await contractFactory.getClassHash();
+        const constructorCalldata = [BigInt(this.publicKey).toString()];
+
+        const maxFee = numericToHexString(0);
+        const nonce = numericToHexString(0);
+
+        const calldataHash = hash.computeHashOnElements([
+            classHash,
+            this.salt,
+            ...constructorCalldata
+        ]);
+        const msgHash = hash.computeHashOnElements([
+            TransactionHashPrefix.DEPLOY_ACCOUNT,
+            numericToHexString(QUERY_VERSION),
+            this.address,
+            0, // entrypoint selector is implied
+            calldataHash,
+            maxFee,
```

Comment:
Does estimation really require a maxFee? Or is some other value expected to be zero here?

Reply:
Yes, I think I linked to the wrong function in the prev comment, but if you see the `estimateAccountDeployFee()` [here](https://github.com/0xs34n/starknet.js/blob/47ffeb6a611eb425482741aaca27aa444a9cd0ea/src/account/default.ts#L196) the estimation also passes a Zero `maxFee` to get a signature. And to clear things out, the `estimateDeployAccountFee()` is looking at the estimation.